exotic/plate_status.py

```diff
@@ -0,0 +1,99 @@
+            self.statusByFilename[filename] = {}
+        self.filename = filename
+        return self
+    # Log an error
+    def _logError(self, errorcode: str, message: str) -> None:
+        if self.filename not in self.statusByFilename:
+            self.statusByFilename[self.filename] = {}
+        rec = self.statusByFilename[self.filename]
+        if errorcode in rec:
+            return
+        # Mark error on this file
+        rec[errorcode] = True
+        self.errorcodes.add(errorcode)
+        # And log new warning
+        self.logfunc(message, warn=True)
+    # Report out of frame warning for start ;index' (0=target, 1+=comp #N)
+    def outOfFrameWarning(self, starIndex):
+        if starIndex == 0:  # Target star
+            self._logError("outofframe_target",
+                f"Target star beyond edge of file {self.filename}")
+        else:
+            self._logError(f"outofframe_comp{starIndex}",
+                f"Comparison star #{starIndex} star beyond edge of file {self.filename}")
+    # Report low flux amplitude warning for start ;index' (0=target, 1+=comp #N)
+    def lowFluxAmplitudeWarning(self, starIndex: int, xc: float, yc: float):
+        if starIndex == 0:  # Target star
+            self._logError("lowflux_target",
+                f"Measured flux for Target star is low in file  {self.filename} - are you sure there is a star at [{xc:.1f}, {yc:.1f}]?")
+        else:
+            self._logError(f"lowflux_comp{starIndex}",
+                f"Measured flux for Comparison star #{starIndex} is low in file {self.filename} - are you sure there is a star at [{xc:.1f}, {yc:.1f}]?")
+    # Report sky background warning for start ;index' (0=target, 1+=comp #N)
+    def skyBackgroundWarning(self, starIndex: int, xc: float, yc: float):
+        if starIndex == 0:  # Target star
+            self._logError("skybg_target",
+                f"Sky background error for Target star for file {self.filename} - are you sure there is a star at [{xc:.1f}, {yc:.1f}]?")
+        else:
+            self._logError(f"skybg_comp{starIndex}",
+                f"Sky background error for Comparison star #{starIndex} for file {self.filename} - are you sure there is a star at [{xc:.1f}, {yc:.1f}]?")
+    # Reort file format error
+    def fitsFormatError(self, e: OSError):
+        self._logError("fits_error",
+            f"Corrupted file {self.filename} ({e}) --removed from reduction")
+    # Reort alignment error
+    def setObsTime(self, time):
+        if self.filename not in self.statusByFilename:
+            self.statusByFilename[self.filename] = {}
+        self.statusByFilename[self.filename]['time'] = time
+        return self
+    # Reort frame time
+    def alignmentError(self):
+        self._logError("alignment_error",
+            f"File {self.filename} failed to align with first file")
+    # Write plate status to CSV file
+    def writePlateStatus(self, file: str):
+        with open(file, 'w') as f:
```

Comment:
I'm trying to use Pathlib for most things nowadays, but okay.

Reply:
See former comment - net-net, this is consistent with other file output code patterns versus what I personally might prefer, and the PR attempts to follow pattern versus prescribe changes that would only be inconsistent with existing code unless the change altered all of them, as well.

Reply:
✔️